### ISSUE TITLE
Class method with a parameter of the same name as the method's doesn't work on Safari

### ISSUE THREAD

**Issue by shuhei** [babel/babel#1456]
I know this is a non-spec behavior of Safari but...

```js
class Foo {
  t(t) {
    return t;
  }
}
```

```
SyntaxError: Cannot declare a parameter named 't' as it shadows the name of a strict mode function.
```

According to https://github.com/babel/babel/issues/1121, function expression works well with the Safari's behavior. Is there any chance that class method will be supported too?

```
$ babel --version
5.2.15
$ echo "var t = function(t) { return t; };" | babel -r
'use strict';

var t = function t(_t) {
  return _t;
};
$ echo "class Foo { t(t) { return t; } }" | babel -r
'use strict';

var Foo = (function () {
  function Foo() {
    babelHelpers.classCallCheck(this, Foo);
  }

  babelHelpers.createClass(Foo, [{
    key: 't',
    value: function t(t) {
      return t;
    }
  }]);
  return Foo;
})();
```

**Comment by monsanto** [comment]
Perhaps like the new `jscript` transformer, we should have a separate `safari` transformer that works around this behavior? Would let us remove [this hack](https://github.com/babel/babel/blob/a241300ff1a817d2e495de949ad15b5f22c44e7c/src/babel/transformation/helpers/name-method.js#L66) (which apparently doesn't work in all cases given this issue), and would have the advantage of fixing functions that aren't named by Babel as well

**Comment by sebmck** [comment]
Weird, should be getting handled, [this](https://github.com/babel/babel/blob/master/src/babel/transformation/helpers/name-method.js#L67-78) comment explains a bit.

**Comment by sebmck** [comment]
Weird, this is being handled correctly:

```javascript
var g = { t: function(t) { return t; } };
```

```javascript
var g = { t: function t(_t) {
    return _t;
  } };
```

**Comment by shuhei** [comment]
@monsanto Thanks for your comment. It would be nice for internal design to extract a transformer for Safari hack but IMHO it should not be optional because Safari (expecially mobile one) has large user base.

@sebmck Thank you very much for the fix!

**Comment by sebmck** [comment]
Fixed as of 5.2.16, thanks!
